yoko-testify/src/test/java/testify/hex/HexBuilderTest.java

```diff
@@ -0,0 +1,151 @@
+        String hex = "abc";
+        assertThrows(AssertionError.class, () -> newHexBuildObject.seq(hex));
+    }
+
+    @Test
+    void testSeqHexInvalidExpr() {
+        HexBuilder newHexBuildObject = HexBuilder.buildHex();
+        String hex = "dq";
+        assertThrows(AssertionError.class, () -> newHexBuildObject.seq(hex));
+    }
+
+    @Test
+    void testSeqHex() {
+        HexBuilder newHexBuildObject = HexBuilder.buildHex();
+        String hex = "ac12118c";
+        String expected = "00000004ac12118c";
+        String result = newHexBuildObject.seq(hex).hex();
+        assertEquals(expected, result);
+    }
+
+    @Test
+    void testSeqBytes() {
+        HexBuilder newHexBuildObject = HexBuilder.buildHex();
+        byte[] byte_array = {10, 5, 25, 100, 127};
+        String expected = "000000050a0519647f";
+        String result = newHexBuildObject.seq(byte_array).hex();
+        assertEquals(expected, result);
+    }
+
+    @Test
+    void testNonAsciiStr() {
+        HexBuilder newHexBuildObject = HexBuilder.buildHex();
+        String s = "∞¶#";
+        assertThrows(AssertionError.class, () -> newHexBuildObject.str(s));
+    }
+
+    @Test
+    void testAsciiCharStr() {
+        HexBuilder newHexBuildObject = HexBuilder.buildHex();
+        String s = "B";
+        String expected = "000000024200";
+        String result = newHexBuildObject.str(s).hex();
+        assertEquals(expected, result);
+    }
+
+    @Test
+    void testAsciiStr() {
+        HexBuilder newHexBuildObject = HexBuilder.buildHex();
+        String s = "This message is a secret!";
+        String expected = "0000001a54686973206d6573736167652069732061207365637265742100";
+        String result = newHexBuildObject.str(s).hex();
+        assertEquals(expected, result);
+    }
+
+    @Test
+    void testUnfinishedCdrEncapsulation() {
+        HexBuilder newHexBuildObject = HexBuilder.buildHex();
+        HexBuilder hexBuilderCapsule = newHexBuildObject.cdr();
+        assertThrows(IllegalStateException.class, hexBuilderCapsule::hex);
+    }
+
+    @Test
+    void testEmptyCdrEncapsulation() {
+        HexBuilder newHexBuildObject = HexBuilder.buildHex();
+        HexBuilder hexBuilderCapsule = newHexBuildObject.cdr();
+        String expected = "0000000100";
+        String result = hexBuilderCapsule.end().hex();
+        assertEquals(expected, result);
+    }
+
+    @Test
+    void testCdrEncapsulation() {
+        HexBuilder newHexBuildObject = HexBuilder.buildHex();
+        int[] listInt = {12, 13, 100, 128};
+        String expected = "0c0d64800000000100";
+        newHexBuildObject.oct(listInt);
+        HexBuilder hexBuilderCapsule = newHexBuildObject.cdr();
+        String result = hexBuilderCapsule.end().hex();
+        assertEquals(expected, result);
+    }
+
+    @Test
+    void testReturnHexString() {
+        HexBuilder newHexBuildObject = HexBuilder.buildHex();
+        assertEquals(newHexBuildObject.hex().getClass(), String.class);
+    }
+}
```

Comment:
Add a trailing newline at the end of the file.

Reply:
Solved